tests/configlet/util/generic_patch.py

```diff
@@ -0,0 +1,217 @@
+        duthost.copy(src=os.path.join(patch_add_t0_dir, fl), dest=sonic_fl)
+        res = duthost.shell(CMD.format(sonic_fl), module_ignore_errors=True)
+
+        # HACK: TODO: There are scenarios, where it applies patch and still consider as failed
+        # "...Error: After applying patch to config, there are still some parts not updated\n"
+        # The above error is possible, if some control plane component is making an update.
+        # We can ignore rc, as DB comp is the final check. So skip it.
+        # assert res["rc"] == 0, "Failed to apply patch"
+
+    do_pause(PAUSE_CLET_APPLY, "Pause after applying add patch")
+
+    if hack_apply:
+        duthost.shell("config interface startup {}".format(tor_ifname))
+        do_pause(PAUSE_INTF_UP, "pause upon i/f {} startup after add patch".format(tor_ifname))
+
+    assert wait_until(DB_COMP_WAIT_TIME, 20, 0, db_comp, duthost, patch_add_t0_dir,
+            orig_db_dir, "generic_patch_add_t0"), \
+            "DB compare failed after adding T0 via generic patch updater"
+
+    # Ensure BGP session is up
+    chk_bgp_session(duthost, tor_data["ip"]["remote"], "post-patch-add test")
+    chk_bgp_session(duthost, tor_data["ipv6"]["remote"].lower(), "post-patch-add test")
+
+
+def generic_patch_rm_t0(duthost, skip_load=False, hack_apply=False):
+    # Load config with T0
+    #
+    if not skip_load:
+        duthost.copy(src=os.path.join(orig_db_dir, "config_db.json"),
+                dest="/etc/sonic/config_db.json")
+        config_reload(duthost, wait=RELOAD_WAIT_TIME, start_bgp=True)
+
+    if hack_apply:
+        # Hack: TODO: Before removing port, patch updater need to ensure
+        # the port is down. Until then bring it down explicitly.
+        #
+        tor_ifname = tor_data["links"][0]["local"]["sonic_name"]
+        duthost.shell("config interface shutdown {}".format(tor_ifname))
+        do_pause(PAUSE_INTF_DOWN, "pause upon i/f {} shutdown before add patch".format(tor_ifname))
+
+    patch_files = _list_patch_files(patch_rm_t0_dir)
+
+    CMD = CMD_APPLY_HACK if hack_apply else CMD_APPLY
+
+    for fl in patch_files:
+        sonic_fl = os.path.join("/etc/sonic", fl)
+        duthost.copy(src=os.path.join(patch_rm_t0_dir, fl), dest=sonic_fl)
+        res = duthost.shell(CMD.format(sonic_fl), module_ignore_errors=True)
+
+        # HACK: TODO: There are scenarios, where it applies patch and still consider as failed
+        # "...Error: After applying patch to config, there are still some parts not updated\n"
+        # The above error is possible, if some control plane component is making an update.
+        # We can ignore rc, as DB comp is the final check. So skip it.
+        # assert res["rc"] == 0, "Failed to apply patch"
+
+    assert wait_until(DB_COMP_WAIT_TIME, 20, 0, db_comp, duthost, patch_rm_t0_dir,
+            no_t0_db_dir, "generic_patch_rm_t0"), \
+            "DB compare failed after adding T0 via generic patch updater"
+
```

Comment:
I think we should test config rollback once the `hack` is removed.

Reply:
yes